patches/server/0255-Improve-death-events.patch

```diff
@@ -296,9 +292,46 @@ index c47884bab387647d99ed842b46f1c078ef9b6995..a3f3e06679cef10f50346e9cc6672ec9
      public void addAdditionalSaveData(CompoundTag nbt) {
          super.addAdditionalSaveData(nbt);
 diff --git a/src/main/java/net/minecraft/world/entity/decoration/ArmorStand.java b/src/main/java/net/minecraft/world/entity/decoration/ArmorStand.java
-index f94a74728bd7c02a7f8245c92e7916f0b669ee0d..cd54fa8f7bbcb6036e90f4ef7cdc01d7af835a13 100644
+index f94a74728bd7c02a7f8245c92e7916f0b669ee0d..1c527ee717c70d1448721da21b8d2bc263302817 100644
 --- a/src/main/java/net/minecraft/world/entity/decoration/ArmorStand.java
 +++ b/src/main/java/net/minecraft/world/entity/decoration/ArmorStand.java
+@@ -531,9 +531,9 @@ public class ArmorStand extends LivingEntity {
+                             this.gameEvent(GameEvent.ENTITY_DAMAGE, source.getEntity());
+                             this.lastHit = i;
+                         } else {
+-                            this.brokenByPlayer(source);
++                            org.bukkit.event.entity.EntityDeathEvent event = this.brokenByPlayer(source); // Paper
+                             this.showBreakingParticles();
```

Comment:
why show the particles if the event is canceled

Reply:
For consistency, mostly, the non-player death causes all show particles when cancelled. The other death causes that show the particles don't always have an easily accessible way to tell if the event was cancelled, so this was the easiest solution.